### PR TITLE
fix(ext/node): implement process.kill in Rust

### PR DESCRIPTION
**Pull request by littledivy**
Closes https://github.com/denoland/deno/issues/23056

### CODE REVIEW

File: ext/node/ops/process.rs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    if TerminateProcess(p_hnd, 1) == TRUE {
+      return 0;
+    }
+
+    GetLastError()
+  }
+}
+
+#[op2(fast)]
+pub fn op_node_process_kill(#[smi] pid: i32, #[smi] sig: i32) -> i32 {
```
Comment (littledivy) [inline ext/node/ops/process.rs:47]: TODO: permission check

File: tests/unit_node/process_test.ts [hunk 2]
```diff
@@ -235,6 +235,19 @@ Deno.test({
   },
 });
 
+Deno.test({ permissions: { run: true, read: true } }, function processKill() {
+  // deno-lint-ignore no-deprecated-deno-api
+  const p = Deno.run({
```
Comment (satyarohith) [inline tests/unit_node/process_test.ts:240]: why not use Deno.Command?

Comment (littledivy) [reply tests/unit_node/process_test.ts:240]: Fixed

**Review by satyarohith — APPROVED** [review]
LGTM

**Pull request merged**